xarray/backends/rasterio_.py

```diff
@@ -38,9 +38,7 @@ def shape(self):
         return self._shape
 
     def __getitem__(self, key):
-
-        # make our job a bit easier
-        key = indexing.canonicalize_indexer(key, self._ndims)
```

Comment:
Agreed, we can get rid of it, it is indeed entirely redundant and unnecessary (all indexing in xarray goes through `Variable.__getitem__` already).